src/tools/ceph_objectstore_tool.cc

```diff
@@ -1981,25 +1981,36 @@ int do_meta(ObjectStore *store, string object, Formatter *formatter, bool debug,
   return 0;
 }
 
+enum rmtype {
+  STANDARD,
```

Comment:
so

- STANDARD for removing both snap from snapmapper and the object
- SNAPMAP for removing the snap from snapmapper only
- NOSNAPMAP for remove the object only

i'd suggest rename `STANDARD` to `BOTH`, `NOSNAPMAP` to `OBJECT`, like
```c++
enum class Remove {
  BOTH,
  SNAPMAP,
  OBJECT
};

and update `remove_object()` like,
```c++
if (type == Remove::BOTH || type == Remove::SNAPMAP) {
  // remove snap from snapmap
}
if (type == Remove::BOTH || type == Remove::OBJECT) {
  // remove the object
}
```

easier to digest this way, IMHO.